src/wp-includes/option.php

```diff
@@ -776,6 +776,24 @@ function update_option( $option, $value, $autoload = null ) {
 	 */
 	$value = apply_filters( 'pre_update_option', $value, $option, $old_value );
 
+	/*
+	 * To get the actual raw old value from the database, any existing pre filters need to be temporarily disabled.
+	 * Immediately after getting the raw value, they are reinstated.
+	 * The raw value is only used to determine whether a value is present in the database. It is not used anywhere
+	 * else, and is not passed to any of the hooks either.
+	 */
+	if ( has_filter( "pre_option_{$option}" ) ) {
+		global $wp_filter;
+
+		$old_filters = $wp_filter[ "pre_option_{$option}" ];
+		unset( $wp_filter[ "pre_option_{$option}" ] );
+
+		$raw_old_value                       = get_option( $option );
+		$wp_filter[ "pre_option_{$option}" ] = $old_filters;
```

Comment:
```suggestion
		$raw_old_value = __get_option( $option );
```

Did you consider / try this?

Reply:
@spacedmonkey This is a private function, and we would have to include a WP Admin file for it, so I think we should avoid using that function. It would also introduce other side effects, for example it's missing the `default_option_{$option}` filter too.

Reply:
Could we do something simplier like this.

```
$suppress = $wpdb->suppress_errors();
		$row      = $wpdb->get_row( $wpdb->prepare( "SELECT option_value FROM $wpdb->options WHERE option_name = %s LIMIT 1", $option ) );
		$wpdb->suppress_errors( $suppress );

		if ( is_object( $row ) ) {
			$value = $row->option_value;
		}
```

Reply:
@spacedmonkey That may be tricky in regards to what I replied in my other comment. What about sites that generally modify how options are being retrieved (see `pre_option` filter)?

I think temporarily unhooking only the filter for specific options is a safer solution here. Of course this could always be improved later, since the change here doesn't introduce any new extension points, so we could completely revert it again if we wanted to, without breakage.